compiler/rustc_middle/src/ty/error.rs

```diff
@@ -575,7 +575,7 @@ impl<T> Trait<T> for X {
                 // This will also work for `impl Trait`.
                 let def_id = if let ty::Param(param_ty) = proj_ty.self_ty().kind() {
                     let generics = self.generics_of(body_owner_def_id);
-                    generics.type_param(param_ty, self).def_id
+                    generics.type_param(&param_ty, self).def_id
```

Comment:
Probably want to instead take `ParamTy` by value in `type_param`, but using a reference is fine for now here